go-controller/pkg/node/gateway_init.go

```diff
@@ -88,6 +88,9 @@ func getNetworkInterfaceIPAddresses(iface string) ([]*net.IPNet, error) {
 	for _, addr := range addrs {
 		switch ip := addr.(type) {
 		case *net.IPNet:
+			if ip.IP.IsLinkLocalUnicast() {
```

Comment:
this matches the 169.254.*.* too, that seems used in other places. can this be a problem?

Reply:
yeah would this be better under the ipv6 check @danwinship ?

Reply:
I did it IPv6-only at first and then was like, well, why should we flush IPv4 LL addresses. But I guess that is pretty different...

Reply:
oh wait, this isn't the flush code.

I don't think it makes sense to allow using a link-local IP address as your node IP whether it's IPv4 or IPv6. In particular, if the node needs to communicate with anything off-link, it needs a non-LL IP, and if it doens't need to communicate with anything off-link, you can just use `10.0.0.0/8` or whatever because you don't have to worry about conflicts with IPs used elsewhere.

Reply:
I was worried because of this

```
go-controller/pkg/node/gateway_localnet.go:     v4localnetGatewayIP           = "169.254.33.2"
go-controller/pkg/node/gateway_localnet.go:     v4localnetGatewayNextHop      = "169.254.33.1"
go-controller/pkg/util/subnet_annotations.go://         "default": ["169.254.16.21", "fd99::10:21"]
go-controller/pkg/util/util.go: V4NodeLocalNatSubnet           = "169.254.0.0/20"
go-controller/pkg/util/util.go: V4NodeLocalNatSubnetNextHop    = "169.254.0.1"
go-controller/pkg/util/util.go: V4NodeLocalDistributedGwPortIP = "169.254.0.2"
go-controller/pkg/util/node_annotations.go://           "ip-addresses": ["169.254.33.2/24"],
go-controller/pkg/util/node_annotations.go://           "next-hops": ["169.254.33.1"],
go-controller/pkg/util/node_annotations.go://           "ip-address": "169.254.33.2/24",
go-controller/pkg/util/node_annotations.go://           "next-hop": "169.254.33.1",
```
but since I still not much familiar on how this interact just asked if it can have some conflict

Reply:
yeah we do use 169.254 which is link local. Although maybe linux considers it global cause we are using a /20 instead of a /16? I dont know, either way I would urge on the side of caution here and only mess with ipv6.

Reply:
But this function is called to get the IP of "eth0"/"br-ex"/etc. We don't add link-local IPs to the default network interface, _and if we did, we wouldn't want to return those IPs here_. The intent of this function is to figure out what the official node IP is. In fact, we should probably allow passing it in on the command line so we can just get it from kubelet when ovnkube is being run as a pod.

Reply:
removed that commit. I'll file an issue about adding a command-line flag